rust/arrow/regen.sh

```diff
@@ -0,0 +1,112 @@
+
+# Now the files are wrongly named so we have to change that
+popd
+pushd $DIR/src/ipc/gen
+for f in `ls *_generated.rs`; do
+    adj_length=$((${#f}-13))
+    mv $f "${f:0:$adj_length}.rs"
+done
+
+PREFIX=$(cat <<'HEREDOC'
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#![allow(dead_code)]
+#![allow(unused_imports)]
+
+use std::{cmp::Ordering, mem};
+use flatbuffers::EndianScalar;
+
+HEREDOC
+)
+
+SCHEMA_IMPORT="\nuse crate::ipc::gen::Schema::*;"
+SPARSE_TENSOR_IMPORT="\nuse crate::ipc::gen::SparseTensor::*;"
+TENSOR_IMPORT="\nuse crate::ipc::gen::Tensor::*;"
+
+
+# Remove all generated lines we don't need
+for f in `ls *.rs`; do
+
+    if [[ $f == "mod.rs" ]]; then
+        continue
+    fi
+
+    echo "Modifying: $f"
+    sed -i '' '/extern crate flatbuffers;/d' $f
+    sed -i '' '/use self::flatbuffers::EndianScalar;/d' $f
+    sed -i '' '/\#\[allow(unused_imports, dead_code)\]/d' $f
+    sed -i '' '/pub mod org {/d' $f
+    sed -i '' '/pub mod apache {/d' $f
+    sed -i '' '/pub mod arrow {/d' $f
+    sed -i '' '/pub mod flatbuf {/d' $f
+    sed -i '' '/}  \/\/ pub mod flatbuf/d' $f
+    sed -i '' '/}  \/\/ pub mod arrow/d' $f
+    sed -i '' '/}  \/\/ pub mod apache/d' $f
+    sed -i '' '/}  \/\/ pub mod org/d' $f
+    sed -i '' '/use std::mem;/d' $f
+    sed -i '' '/use std::cmp::Ordering;/d' $f
+
+    # Replace all occurrences of type__ with type_
+    sed -i '' 's/type__/type_/g' $f
+
+    # Some files need prefixes
+    if [[ $f == "File.rs" ]]; then 
+        # Now prefix the file with the static contents
+        echo -e "${PREFIX}" "${SCHEMA_IMPORT}" | cat - $f > temp && mv temp $f
+    elif [[ $f == "Message.rs" ]]; then
+        echo -e "${PREFIX}" "${SCHEMA_IMPORT}" "${SPARSE_TENSOR_IMPORT}" "${TENSOR_IMPORT}" | cat - $f > temp && mv temp $f
+    elif [[ $f == "SparseTensor.rs" ]]; then
+        echo -e "${PREFIX}" "${SCHEMA_IMPORT}" "${TENSOR_IMPORT}" | cat - $f > temp && mv temp $f
+    elif [[ $f == "Tensor.rs" ]]; then
+        echo -e "${PREFIX}" "${SCHEMA_IMPORT}" | cat - $f > temp && mv temp $f
+    else
+        echo "${PREFIX}" | cat - $f > temp && mv temp $f
+    fi
+done
+
+
+
+for f in `ls *.rs`; do
+    echo "Formatting: $f"
+    rustfmt $f
```

Comment:
I'm wondering if `cargo +stable fmt` would be better here. Looking at your commited files, it appears that they haven't been formatted 'correctly'. I'd have expected the only changes to be the style of the imports, with everything else being the same (except for the autogenerated warning comment).

I'm on Windows without WSL, so I can't run the bash script myself to confirm.

Reply:
This is slightly tricky. In CI we run `cargo +stable fmt --all -- --check` to confirm that the source code in the PR is formatted correctly, so running `cargo +stable fmt` would defeat the point of that because it would format all the code in the crate (not just these generated files) and then the check would pass.

It would be nice if we could just have rustfmt ignore these generated files. I had attempted to do that through the `rustfmt.toml` in the repo that attempts to ignore these files but we see the warning:
Warning: can't set `ignore = IgnoreList { path_set: {"arrow/src/ipc/gen"}, rustfmt_toml_path: "" }`, unstable features are only available in nightly channel.
```
This confuses me slightly because we _are_ using the nightly channel.

```

Reply:
Perhaps running it with nightly on the bash script is better than my `stable` suggestion. We wouldn't run the script in CI, so we wouldn't get the fmt issues

Reply:
I think, I addressed this by adding an additional option to the script. Right now, I will execute this as

```
cargo +stable fmt -- src/ipc/gen/*
```

and this will make 

```
cargo +stable fmt --all -- --check
```

pass without issues.